src/features/categories/category-selection/CategorySelect.tsx

```diff
@@ -31,52 +33,62 @@ const CategorySelect = ({
     setIsOpen((prev) => !prev);
   };
 
-  const handleCategorySelection = (categoryId: IndexableType, operation: 'add' | 'remove') => {
-    switch (operation) {
-      case 'add':
-        setSelectedCategories([...categoryIds, categoryId]);
-        break;
-      case 'remove':
-        setSelectedCategories(categoryIds.filter((id) => id !== categoryId));
-        break;
-    }
-  };
+const handleCategorySelection = (categoryId: IndexableType, operation: 'add' | 'remove') => {
+  if (readOnly && operation === 'remove') return;
+
+  switch (operation) {
+    case 'add':
+      setSelectedCategories([...categoryIds, categoryId]);
+      break;
+    case 'remove':
+      setSelectedCategories(categoryIds.filter((id) => id !== categoryId));
+      break;
+  }
+};
 
   return (
     <>
-      <div className={classes.categorySelect}>
-        <span className={classes.label}>{label}</span>
-        <div
-          tabIndex={0}
-          className={cx({ select: true, isOpen })}
-          onClick={handleSelectToggle}
-        >
-          <div className={classes.selectIcon}>
-            <Hash />
-          </div>
-          <div className={classes.selectedCategoriesWrapper}>
-            {categoryIds.length === 0 ? (
-              <span className={classes.placeholder}>{placeholder}</span>
-            ) : (
-              <div className={classes.flexContainer}>
-                {categories
-                  ?.filter(({ id }) => categoryIds.includes(id))
-                  .map((cat) => (
-                    <div className={classes.flexContainerItem} key={cat.id.toString()}>
-                      <CategoryBadge
-                        {...cat}
-                        onClick={() => handleCategorySelection(cat.id, 'remove')}
-                      />
-                    </div>
-                  ))}
+   <div className={classes.categorySelect}>
+  <span className={classes.label}>{label}</span>
+  <div
+    tabIndex={0}
+    className={cx({ select: true, isOpen })}
+    onClick={handleSelectToggle}
+  >
+    <div className={classes.selectIcon}>
+      <Hash />
+    </div>
+    <div className={classes.selectedCategoriesWrapper}>
+      {categoryIds.length === 0 ? (
+        <span className={classes.placeholder}>{placeholder}</span>
+      ) : (
+        <div className={classes.flexContainer}>
+          {categories
+            ?.filter(({ id }) => categoryIds.includes(id))
+            .map((cat) => (
+              <div className={classes.flexContainerItem} key={cat.id.toString()}>
+               <CategoryBadge
+  {...cat}
+  onClick={!readOnly ? () => handleCategorySelection(cat.id, 'remove') : undefined}
+/>
```

Comment:
PLease format this document using `shift`+`alt`+`f`. The indentations got messed up.